submissions/m-ruslan/oop-exercise/index.js

```diff
@@ -0,0 +1,192 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/m-ruslan/a-tiny-JS-world
+   Web app: https://m-ruslan.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+//**ES6 classes and sub-classes
+class Inhabitant {
+  constructor(name, legs, gender, propNamesArr) {
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.friendTo = [];
+    this.propNamesArr = propNamesArr;
```

Comment:
I see the problem you're trying to resolve by holding `propNamesArr`.
Not illegal, yet not really elegant.